lib/ecto/adapters/mysql/connection.ex

```diff
@@ -0,0 +1,534 @@
+    end
+
+    defp column_definition({:add, name, %Reference{} = ref, opts}) do
+      # The foreign key time must match. Check if there is a way to get
+      # the referenced key's type
+      assemble([quote_name(name), "BIGINT UNSIGNED", column_options(name, opts), reference_expr(ref, name)])
+    end
+
+    defp column_definition({:add, name, type, opts}) do
+      assemble([quote_name(name), column_type(type, opts), column_options(name, opts)])
+    end
+
+    defp column_changes(columns) do
+      Enum.map_join(columns, ", ", &column_change/1)
+    end
+
+    defp column_change({:add, name, type, opts}) do
+      assemble(["ADD", quote_name(name), column_type(type, opts), column_options(name, opts)])
+    end
+
+    defp column_change({:modify, name, type, opts}) do
+      assemble(["MODIFY", quote_name(name), column_type(type, opts)])
+    end
+
+    defp column_change({:remove, name}), do: "DROP #{quote_name(name)}"
+
+    defp column_options(name, opts) do
+      default = Keyword.get(opts, :default)
+      null    = Keyword.get(opts, :null)
+      pk      = Keyword.get(opts, :primary_key)
+      unique  = Keyword.get(opts, :unique)
+
+      [default_expr(default), unique_expr(unique), null_expr(null), pk_expr(pk, name)]
+    end
+
+    defp pk_expr(true, name), do: ", PRIMARY KEY(#{quote_name(name)})"
+    defp pk_expr(_, _), do: nil
+
+    defp unique_expr(true), do: "UNIQUE"
+    defp unique_expr(_), do: nil
+
+    defp null_expr(false), do: "NOT NULL"
+    defp null_expr(_), do: nil
+
+    defp default_expr(nil),
+      do: nil
+    defp default_expr(literal) when is_binary(literal),
+      do: "DEFAULT '#{escape_string(literal)}'"
+    defp default_expr(literal) when is_number(literal) or is_boolean(literal),
+      do: "DEFAULT #{literal}"
+    defp default_expr({:fragment, expr}),
+      do: "DEFAULT #{expr}"
+
+    defp index_expr(literal), do: quote_name(literal)
+
+    defp reference_expr(%Reference{} = ref, foreign_key_name) do
+      ", FOREIGN KEY (#{quote_name(foreign_key_name)}) REFERENCES #{quote_name(ref.table)} (#{quote_name(ref.column)})"
+    end
+
+    defp column_type(type, opts) do
+      size      = Keyword.get(opts, :size)
+      precision = Keyword.get(opts, :precision)
+      scale     = Keyword.get(opts, :scale)
+      type_name = ecto_to_db(type)
+
+      cond do
+        size            -> "#{type_name}(#{size})"
+        precision       -> "#{type_name}(#{precision},#{scale || 0})"
+        type == :string -> "#{type_name}(255)"
+        true            -> "#{type_name}"
+      end
+    end
+
+    ## Helpers
+
+   defp quote_name(name), do: "`#{name}`"
+
+    defp assemble(list) do
+      list
+      |> List.flatten
+      |> Enum.filter(&(&1 != nil))
+      |> Enum.join(" ")
+    end
+
+    defp escape_string(value) when is_binary(value) do
+      :binary.replace(value, "'", "''", [:global])
+    end
+
+    defp ecto_to_db(:string),     do: "varchar"
+    defp ecto_to_db(:datetime),   do: "datetime"
```

Comment:
Floats in Elixir are 64 bit double precision and MySQL uses four bytes for single-precision values (Float) and eight bytes for double-precision (Double) values.

http://elixir-lang.org/getting-started/basic-types.html
http://dev.mysql.com/doc/refman/5.0/en/floating-point-types.html

Shall we save the `Float` type in Elixir as `Double` in MySQL in order to keep precision?